bluebottle/recurring_donations/management/commands/process_monthly_donations.py

```diff
@@ -85,3 +84,282 @@ def handle(self, *args, **options):
+                logger.info("--")
+
+
+def _process_monthly_order(monthly_order, send_email=False):
+    if monthly_order.processed:
+        logger.info(
+            "Order for {0} already processed".format(monthly_order.user))
+        return False
+
+    ten_days_ago = timezone.now() + timezone.timedelta(days=-10)
+    recent_orders = Order.objects.filter(user=monthly_order.user,
+                                               order_type='recurring',
+                                               updated__gt=ten_days_ago)
+
+    if recent_orders.count() > 0:
+        message = "Skipping '{0}' recently processed a recurring order for {1}:".format(
+            monthly_order, monthly_order.user)
+        logger.warn(message)
+        for closed_order in recent_orders.all():
+            logger.warn("Recent Order Number: {0}".format(closed_order.id))
+
+        # Set an error on this monthly order
+        monthly_order.error = message
+        monthly_order.save()
+        return False
+
+    order = Order.objects.create(status=StatusDefinition.LOCKED,
+                                       user=monthly_order.user,
+                                       order_type='recurring')
+    order.save()
+
+    logger.info(
+        "Creating Order for {0} with {1} donations".format(monthly_order.user,
+                                                           monthly_order.donations.count()))
+    for monthly_donation in monthly_order.donations.all():
+        donation = Donation.objects.create(amount=monthly_donation.amount,
+                                           project=monthly_donation.project,
+                                           order=order)
+        donation.save()
+
+    integration_data = {'account_name': monthly_order.name,
+                        'account_city': monthly_order.city,
+                        'iban': monthly_order.iban,
+                        'bic': monthly_order.bic,
+                        'agree': True}
+
+    order_payment = OrderPayment(order=order, user=monthly_order.user,
+                                 payment_method=PAYMENT_METHOD,
+                                 integration_data=integration_data)
+
+    order_payment.save()
+
+    try:
+        service = PaymentService(order_payment)
+        service.start_payment()
+    except PaymentException as e:
+        order_payment.delete()
+        order.delete()
+        error_message = "Problem starting payment. {0}".format(e)
+        monthly_order.error = "{0}".format(e.message)
+        monthly_order.save()
+        logger.error(error_message)
+        return False
+
+    logger.debug("Payment for '{0}' started.".format(monthly_order))
+
+    monthly_order.processed = True
+    monthly_order.error = ''
+    monthly_order.save()
+
+    # Try to update status
+    service.check_payment_status()
+
+    # Send an email to the user.
+    if send_email:
+        mail_monthly_donation_processed_notification(monthly_order)
+
+    return True
+
+
+
```

Comment:
I think this is incorrectly merged. All these methods have been moved to tasks.py now.